stdpopsim/cache.py

```diff
@@ -43,3 +49,128 @@ def get_cache_dir():
 
 
 set_cache_dir()
+
+
+@attr.s(kw_only=True)
+class CachedData:
+    """
+    Downloadable data that will be cached locally.
+
+    The downloadable should be a single file. The local cache may
+    correspond to this same file, or to its extracted contents. In the latter
+    case, the downloaded file will be removed after archive extraction.
+    The downloaded file is compared against the expected SHA256 checksum,
+    and if correct, the checksum is then also stored locally.
+
+    :ivar namespace: The namespace under which the cache will be stored.
+        This will be converted into a folder, by constructing folders in the
+        cache corresponding to each component of the namespace.
+        E.g. if we're on a unix system with cache under ``/path/to/cache``, and
+        ``namespace=pathlib.PurePath("foo", "bar")``, the cached data will
+        live under ``/path/to/cache/foo/bar``.
+    :vartype namespace: :class:`pathlib.PurePath`
+    :ivar str url: The URL of the data to be cached.
+    :ivar str sha256: The SHA256 checksum of the downloaded file.
+    :ivar bool extract: True if the downloaded file is a tarball that should be
```

Comment:
This is very useful right now, but I wonder if we should aim to deprecate the hapmap formatted files and package the maps up as zarrs also. Hapmap is a pretty stupid format, as it turns out. Just something to keep in mind: do we want tarballs to be something we support long-term, or do we want to phase this functionality out?

Reply:
Sure. I'd rather not go down that road as part the present PR though. I don't anticipate exposing this new class as a user API though, so we should be able to simplify it to only do the `extract=False` case.

Reply:
SGTM